lib/service/businessLogic.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const kinveyCompletionHandler = require('./kinveyCompletionHandler');
+const kinveyErrors = require('kinvey-datalink-errors');
+
+let registeredFunctions = {};
+let resolve = null;
+
+exports.register = (taskName, functionToExecute) => {
```

Comment:
Better es6 pattern:  define functions and then exports at bottom of file.  This allows for function names in stack traces and taking advantage of better performance of the function declaration.  e.g.

```
function register(taskName, functionToExecute) {

}
....
exports.register = register;